cli/args.go

```diff
@@ -360,15 +366,31 @@ func parseIAMRoleOptions(args []string) (*options.IAMRoleOptions, error) {
 	return optsOut, nil
 }
 
+// getOptionArgIfMatched returns the value to use for the option key, if the arg matches the option. This will handle
+// options passed in as --OPTION_KEY VALUE or --OPTION_KEY=VALUE. The second boolean return value indicates if the arg
+// matches the option, returning false if it does not. This will error if it expects a value in the next arg, but the
+// current arg is the end of the input.
+func getOptionArgIfMatched(arg string, nextArg *string, optionName string) (string, bool, error) {
+	optionOnly := fmt.Sprintf("--%s", optionName)
+	if arg == optionOnly {
+		logIfDeprecatedOption(optionName)
+		if nextArg == nil {
+			return "", true, errors.WithStackTrace(ArgMissingValue(optionName))
+		}
+		return *nextArg, true, nil
+	} else if strings.HasPrefix(arg, optionOnly+"=") {
+		logIfDeprecatedOption(optionName)
+		return strings.TrimPrefix(arg, fmt.Sprintf("--%s=", optionName)), true, nil
```

Comment:
This will return `"", true, nil` when the `arg` is something like `--OPTION_KEY=` with no value after it, meaning no `nextArg`.

Reply:
So it's an error to say `--OPTION_KEY` (with no value after) but it's not an error to say `--OPTION_KEY=` (with no value after).

Reply:
Yes that is correct and intentional. Most CLI will interpret `--OPTION_KEY=` as setting the value to `""`, and that is the approach taken here. This is because of bashism, where `--OPTION_KEY=""` is not the same as `--OPTION_KEY='""'` due to automatic string concatenation (it reduces to `--OPTION_KEY=`).

Reply:
Ah, interesting, thank you for that!

Reply:
Right, I did some reading up on bash, and `""` will get chomped to nothing and `'""'` will become `""`.

Reply:
So if someone actually means to pass in the non-empty string `""`, they should provide it quoted as in `'"'"`?

Reply:
> So if someone actually means to pass in the non-empty string "", they should provide it quoted as in '""'?

Yup that's right.